dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -403,19 +404,27 @@ URL findClassLocation(String className) throws ClassNotFoundException {
     return javaClasses.build();
   }
 
+  /** Returns true if two class names (binary name JLS 13.1) have the same package. */
+  static boolean classesInSamePackage(String classNameA, String classNameB) {
+    // Because package name cannot have '.' at the beginning, we can use lastDotIndex=0 (that will
```

Comment:
This is fine.

Alternatively, you could just add 1, so that the substring includes the dot after the package if there is one, and is the empty string for the default package.

`int lastDotIndexA = classNameA.lastIndexOf('.') + 1;`

Reply:
That's also nice.